src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1560,6 +1560,42 @@ uct_ib_mlx5_devx_reg_exported_key(uct_ib_md_t *ib_md, uct_ib_mem_t *ib_memh)
 #endif
 }
 
+static ucs_status_t
+uct_ib_mlx5_devx_reg_exported_key(uct_ib_md_t *ib_md, uct_ib_mem_t *ib_memh)
+{
+    uct_ib_mlx5_md_t *md    = ucs_derived_of(ib_md, uct_ib_mlx5_md_t);
+    uct_ib_mlx5_mem_t *memh = ucs_derived_of(ib_memh, uct_ib_mlx5_mem_t);
+    ucs_status_t status;
+
+    ucs_assertv(memh->super.exported_lkey == UCT_IB_INVALID_MKEY,
+                "memh %p exported_lkey 0x%" PRIx32, memh,
+                memh->super.exported_lkey);
+    if (!(md->flags & UCT_IB_MLX5_MD_FLAG_INDIRECT_XGVMI)) {
+        return uct_ib_mlx5_devx_xgvmi_umem_mr(md, memh);
```

Comment:
maybe "goto out_umem_mr" and add the label in line 1595
to unite the exit flow in case of umem